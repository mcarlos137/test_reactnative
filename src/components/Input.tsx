import React from 'react';
import { View, StyleSheet, TextInput, Text } from 'react-native';

const Input = ({ label, value, onChangeText, inputMode }: { label: string, value: string, onChangeText: ((text: string) => void), inputMode: 'text' | 'email' | 'numeric' }) => {

  return (
    <View
      style={styles.container}
    >
      <Text style={styles.text}>{label}</Text>
      <TextInput
        inputMode={inputMode}
        value={value}
        maxLength={30}
        onChangeText={onChangeText}

        style={styles.input}
      />
    </View>
  );
}

export default React.memo(Input)

const styles = StyleSheet.create({
  container: {
    borderColor: '#CDD7DD',
    borderWidth: 1,
    padding: 10,
    width: 299,
    height: 53,
    marginTop: 20,
    borderRadius: 8
  },
  text: {
    fontSize: 10,
    color: '#8F979C',
    fontWeight: 500,
    marginBottom: 3,
    fontFamily: 'Roboto'
  },
  input: {
    fontSize: 16,
    color: '#092333',
    fontWeight: 400,
    fontFamily: 'Roboto'
  },
});
