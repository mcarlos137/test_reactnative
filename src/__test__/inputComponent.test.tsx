import * as React from 'react';
import renderer from 'react-test-renderer';
//COMPONENTS
import Input from '../components/Input';


it(`input component renders correctly`, () => {
  const tree = renderer.create(<Input value='Carlos' label='First Name' onChangeText={(text: string) => text} inputMode='text' />).toJSON();
  expect(tree).toMatchSnapshot();
});

