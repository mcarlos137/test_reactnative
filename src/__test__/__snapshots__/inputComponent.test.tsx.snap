// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component renders correctly 1`] = `
<View
  style={
    {
      "borderColor": "#CDD7DD",
      "borderRadius": 8,
      "borderWidth": 1,
      "height": 53,
      "marginTop": 20,
      "padding": 10,
      "width": 299,
    }
  }
>
  <Text
    style={
      {
        "color": "#8F979C",
        "fontFamily": "Roboto",
        "fontSize": 10,
        "fontWeight": 500,
        "marginBottom": 3,
      }
    }
  >
    First Name
  </Text>
  <TextInput
    inputMode="text"
    maxLength={30}
    onChangeText={[Function]}
    style={
      {
        "color": "#092333",
        "fontFamily": "Roboto",
        "fontSize": 16,
        "fontWeight": 400,
      }
    }
    value="Carlos"
  />
</View>
`;
